### DISCUSSION TITLE
Record types

### DISCUSSION THREAD

**Discussion by FelixMalfait** [twentyhq/twenty#4112] [Ideas]
*not ready for implementation*

## Scope & Context
See previous discussion here: https://github.com/twentyhq/twenty/issues/2761
We'll go with Record types (the Salesforce way)

## Implementation
- Create a new table "objectTypesMetadata" with namePlural, labelPlural, description, icon, targetObjectId
- In sidebar, adapt design to display the object types below objects. Upon click on sidebar item, link to the listView but with an additional param in the graphQL query to pass a default objectType=namePlural
- Create a standard field called "objectType" on companies, people, opportunities. We shouldn't be able to delete this field. The frontend implementation is similar to a dropdown list, but the different is that the dropdown values come from the metadata table.
- Add UI in settings:
<img width="540" alt="Screenshot 2023-11-29 at 10 57 48" src="https://github.com/twentyhq/twenty/assets/6399865/0bd7b185-8d7d-425a-aee3-8b19a662724e">
<img width="553" alt="Screenshot 2023-11-29 at 10 58 36" src="https://github.com/twentyhq/twenty/assets/6399865/777af1c1-6685-49b2-bffe-cb5ecd987d67">

[missing "add type" design]

- Adapt views to make them eventually be defined relatively to the objectType
- Adapt fieldMetadata to make them eventually be defined relatively to the objectType

## Other options considered
Introduce parent_id column on objectMetadata and consider that a type is an object, but with a parent from which it inherits a few things

**Comment by BOHEUS** [comment]
@FelixMalfait is it still relevant considering possible customization of record views (changing type of view, showing/hiding fields)?

**Reply by FelixMalfait** [reply]
Probably not indeed. Agee Conditional Fields will help a lot!
